### PR TITLE
Formal parameters sync with literal expression 

### LINKED ISSUE #896: Formal parameters suggestions are not reloaded until view is reopened
### Describe the Bug

Variables are only loaded when you open the BKM, so you need to reopen the view after you add the formal parameters:

https://github.com/user-attachments/assets/2011b103-531a-44c9-be54-571d700ede5c

### Steps to Reproduce

1. Create a BKM
2. Add formal parameters
3. Try to use the suggestions

If you report a modeling related issue, ensure you can reproduce it on [demo.bpmn.io](https://demo.bpmn.io/dmn/new)

When reporting a library error, try to build an example that reproduces your problem. You can use our playgrounds for [viewer](https://jsfiddle.net/x2doq4vn/) or [modeler](https://jsfiddle.net/nrko5hyt/) as a starting point or put a demo up on [GitHub](https://github.com/) for inspection.

### Expected Behavior

Suggestions are loaded immediately.

### Environment

Please complete the following information:

 - Browser: [e.g. IE 11, Chrome 69]
 - OS: [e.g. Windows 7]
 - Library version: [e.g. 2.0.0]

Related to https://github.com/camunda/camunda-modeler/issues/4544
Child of https://github.com/camunda/camunda-modeler/issues/4544

### PR DESCRIPTION
**Pull request by abdul99ahad**
Closes #896 
### Proposed Changes

 

You can update the parameters and the variables will be available in real-time in literal expression.
Nov-05-2024 15-37-49

### Checklist

To ensure you provided everything we need to look at your PR:

* [x] **Brief textual description** of the changes present
* [x] **Visual demo** attached
* [ ] **Steps to try out** present, i.e. [using the `@bpmn-io/sr` tool](https://github.com/bpmn-io/sr)
* [x] Related issue linked via `Closes {LINK_TO_ISSUE}` or `Related to {LINK_TO_ISSUE}`

### CODE REVIEW

**Comment by barmac** [comment]
This is merged now but still needs to be released.

**Comment by abdul99ahad** [comment]
> This is merged now but still needs to be released.

Aright. I'll create the release.

File: packages/dmn-js-boxed-expression/src/features/function-definition/components/FunctionDefinitionEditorComponent.js [hunk 1]
```diff
@@ -102,14 +103,18 @@ function _Parameter({ parameter }) {
...
 
-function BodyExpression({ expression }, context) {
+const BodyExpression = withChangeSupport(_BodyExpression, props => props.parameters);
+
+function _BodyExpression({ expression, parameters }, context) {
   const Expression = context.components.getComponent('expression', {
     expression
   });
 
+  console.log('parameters BodyExpression: ', parameters);
```
Comment (philippfromme) [inline packages/dmn-js-boxed-expression/src/features/function-definition/components/FunctionDefinitionEditorComponent.js:113]: 🤡

Comment (abdul99ahad) [reply packages/dmn-js-boxed-expression/src/features/function-definition/components/FunctionDefinitionEditorComponent.js:113]: Entschuldigung 😄

**Pull request merged**